### ISSUE TITLE
dependency constraint failure by pedump ~> 0.5 on multipart-post, not compatible with signet ~> 0.60. None of which I have the power to change 

### ISSUE THREAD

**Issue by rsb** [chef/omnibus#629]
Bundler could not find compatible versions for gem "multipart-post":
  In Gemfile:
    omnibus (>= 0) ruby depends on
      pedump (~> 0.5) ruby depends on
        multipart-post (~> 1.1.4) ruby

    ap_dfp_api (>= 0) ruby depends on
      google-dfp-api (~> 0.12.1) ruby depends on
        google-ads-common (~> 0.11.0) ruby depends on
          signet (~> 0.6.0) ruby depends on
            faraday (~> 0.9) ruby depends on
              multipart-post (< 3, >= 1.2) ruby

**Comment by bassrock** [comment]
I am see this as well but with this output:

```
Bundler could not find compatible versions for gem "multipart-post":
  In Gemfile:
    berkshelf (>= 3.3) was resolved to 3.3.0, which depends on
      ridley (~> 4.0) was resolved to 4.0.0, which depends on
        faraday (~> 0.9.0) was resolved to 0.9.2, which depends on
          multipart-post (< 3, >= 1.2)

    omnibus was resolved to 5.0.0, which depends on
      pedump (~> 0.5) was resolved to 0.5.0, which depends on
        multipart-post (~> 1.1.4)
```

**Comment by zl4bv** [comment]
Seeing the exact same output as @bassrock. I believe the issue was introduced in 92a08027baffc8aab4742401889086a0103e6a9c.

I managed to work around the issue by relaxing the version constraint on `berkshelf` that was introduced by the [generated](https://github.com/chef/omnibus/blob/master/lib/omnibus/generator_files/Gemfile.erb#L16) Gemfile.

**Comment by bassrock** [comment]
Yep. It makes you make berkshelf go down 2 major versions.

**Comment by rsb** [comment]
I noticed in the Gemfile the patched version of pedump is specified, but in omnibus.gemspec it also appears, so bundler in honoring the gemspec over the Gemfile

**Comment by bassrock** [comment]
It looks like adding:

```ruby
gem 'pedump', git: 'https://github.com/ksubrama/pedump', branch: 'patch-1'
```

to my Gemfile makes it work.

**Comment by rsb** [comment]
good to know, I will try that. I would rather not have to depend on a fork I have to create. Although, I am not pleased on adding a dependency from some developer's fork on outdated dependency, into my production systems. The whole thing seems a little hackish.

**Comment by rsb** [comment]
Yeah that worked.

**Comment by rsb** [comment]
merged into master #630
